### ISSUE TITLE
Missing types in `monitoring` for v6.41.0

### ISSUE THREAD

**Issue by ringods** [pulumi/pulumi#11207]
### What happened?

Customer reports build problems when using v6.41.0 of the Go SDK for this provider:

```
error: error in compiling Go: unable to run `go build`: exit status 2

    # github.com/pulumi/pulumi-gcp/sdk/v6/go/gcp/monitoring
    ../../../../../go/pkg/mod/github.com/pulumi/pulumi-gcp/sdk/v6@v6.41.0/go/gcp/monitoring/service.go:90:15: undefined: ServiceBasicServicePtrOutput
    ../../../../../go/pkg/mod/github.com/pulumi/pulumi-gcp/sdk/v6@v6.41.0/go/gcp/monitoring/service.go:102:14: undefined: ServiceTelemetryArrayOutput
    ../../../../../go/pkg/mod/github.com/pulumi/pulumi-gcp/sdk/v6@v6.41.0/go/gcp/monitoring/service.go:146:16: undefined: ServiceBasicService
    ../../../../../go/pkg/mod/github.com/pulumi/pulumi-gcp/sdk/v6@v6.41.0/go/gcp/monitoring/service.go:158:16: undefined: ServiceTelemetry
    ../../../../../go/pkg/mod/github.com/pulumi/pulumi-gcp/sdk/v6@v6.41.0/go/gcp/monitoring/service.go:171:15: undefined: ServiceBasicServicePtrInput
    ../../../../../go/pkg/mod/github.com/pulumi/pulumi-gcp/sdk/v6@v6.41.0/go/gcp/monitoring/service.go:183:14: undefined: ServiceTelemetryArrayInput
    ../../../../../go/pkg/mod/github.com/pulumi/pulumi-gcp/sdk/v6@v6.41.0/go/gcp/monitoring/service.go:200:16: undefined: ServiceBasicService
    ../../../../../go/pkg/mod/github.com/pulumi/pulumi-gcp/sdk/v6@v6.41.0/go/gcp/monitoring/service.go:222:15: undefined: ServiceBasicServicePtrInput
    ../../../../../go/pkg/mod/github.com/pulumi/pulumi-gcp/sdk/v6@v6.41.0/go/gcp/monitoring/service.go:329:39: undefined: ServiceBasicServicePtrOutput
    ../../../../../go/pkg/mod/github.com/pulumi/pulumi-gcp/sdk/v6@v6.41.0/go/gcp/monitoring/service.go:356:38: undefined: ServiceTelemetryArrayOutput
    ../../../../../go/pkg/mod/github.com/pulumi/pulumi-gcp/sdk/v6@v6.41.0/go/gcp/monitoring/service.go:356:38: too many errors
```

### Steps to reproduce

This file was added to the Go SDK:

https://github.com/pulumi/pulumi-gcp/blob/v6.41.0/sdk/go/gcp/monitoring/service.go

But refers to types not defined elsewhere in the SDK

### Expected Behavior

All types should be generated.

### Actual Behavior

There are 10 missing types.

### Output of `pulumi about`

pulumi-gcp v6.41.0

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by lblackstone** [comment]
@iwahbe pointed out that the `sdk` package wasn't building, so I checked back on older releases to see when this regressed. It appears that https://github.com/pulumi/pulumi-gcp/pull/865 introduced a compilation error.

`v6.32.0` compiles, but `v6.33.0` does not.
```
cd sdk && go build ./...
# github.com/pulumi/pulumi-gcp/sdk/v6/go/gcp/certificatemanager
go/gcp/certificatemanager/certificateMap.go:61:6: CertificateMap redeclared in this block
	go/gcp/certificatemanager/certificate.go:336:6: other declaration of CertificateMap
go/gcp/certificatemanager/certificateMap.go:92:98: cannot use &resource (value of type *CertificateMap) as type pulumi.Resource in argument to ctx.RegisterResource:
	*CertificateMap does not implement pulumi.Resource (missing URN method)
go/gcp/certificatemanager/certificateMap.go:189:6: CertificateMapInput redeclared in this block
	go/gcp/certificatemanager/certificate.go:329:6: other declaration of CertificateMapInput
go/gcp/certificatemanager/certificateMap.go:196:24: CertificateMap.ElementType redeclared in this block
	go/gcp/certificatemanager/certificate.go:338:23: other declaration of ElementType
go/gcp/certificatemanager/certificateMap.go:200:26: CertificateMap.ToCertificateMapOutput redeclared in this block
	go/gcp/certificatemanager/certificate.go:342:25: other declaration of ToCertificateMapOutput
go/gcp/certificatemanager/certificateMap.go:204:26: CertificateMap.ToCertificateMapOutputWithContext redeclared in this block
	go/gcp/certificatemanager/certificate.go:346:25: other declaration of ToCertificateMapOutputWithContext
go/gcp/certificatemanager/certificateMap.go:258:6: CertificateMapOutput redeclared in this block
	go/gcp/certificatemanager/certificate.go:435:6: other declaration of CertificateMapOutput
go/gcp/certificatemanager/certificateMap.go:260:29: CertificateMapOutput.ElementType redeclared in this block
	go/gcp/certificatemanager/certificate.go:437:29: other declaration of ElementType
go/gcp/certificatemanager/certificateMap.go:264:31: CertificateMapOutput.ToCertificateMapOutput redeclared in this block
	go/gcp/certificatemanager/certificate.go:441:31: other declaration of ToCertificateMapOutput
go/gcp/certificatemanager/certificateMap.go:268:31: CertificateMapOutput.ToCertificateMapOutputWithContext redeclared in this block
	go/gcp/certificatemanager/certificate.go:445:31: other declaration of ToCertificateMapOutputWithContext
go/gcp/certificatemanager/certificateMap.go:92:98: too many errors
```

**Comment by iwahbe** [comment]
You can build and test the issue with 
```sh
export PULUMI_EXTRA_MAPPING_ERROR=true;
export PULUMI_MISSING_DOCS_ERROR=true;
export PULUMI_MISSING_MAPPING_ERROR=true;
make tfgen build_go && (cd sdk && go mod tidy && go build ./...);
```

**Comment by iwahbe** [comment]
I think there were two issues with the gcp SDK:
1. We generated duplicate names (fixed by #11244).
2. The SDK didn't delete the old SDK folder during regeneration, leaving orphaned files  from previous generations. (such as `github.com/pulumi/pulumi-gcp/sdk/v6@v6.41.0/go/gcp/monitoring/service.go`) 

By depending on the change from #11244 and running `rm sdk/go` before build, I can build the newly generated schema.
